programs/lz4io.c

```diff
@@ -488,7 +552,12 @@ static int LZ4IO_compressFilename_extRess(cRess_t ress, const char* srcFileName,
     /* multiple-blocks file */
     {
         /* Write Archive Header */
-        size_t headerSize = LZ4F_compressBegin(ctx, dstBuffer, dstBufferSize, &prefs);
+        size_t headerSize;
+        if (ress.cdict) {
```

Comment:
Same idea here : 
`LZ4F_compressBegin_usingCDict()` with `cdict == NULL` is exactly the same as `LZ4F_compressBegin()` : 
https://github.com/lz4/lz4/blob/dev/lib/lz4frame.c#L598

If this property is not clear, maybe it could be added in comment in `lz4frame_static.h`.